clients/client/src/main/java/org/projectnessie/client/util/v2api/ClientSideGetMultipleNamespaces.java

```diff
@@ -40,22 +43,41 @@
  */
 public final class ClientSideGetMultipleNamespaces extends BaseGetMultipleNamespacesBuilder {
   private final NessieApiV2 api;
+  private boolean onlyDirectChildren;
 
   public ClientSideGetMultipleNamespaces(NessieApiV2 api) {
     this.api = api;
   }
 
+  @Override
+  public GetMultipleNamespacesBuilder onlyDirectChildren(boolean onlyDirectChildren) {
+    this.onlyDirectChildren = onlyDirectChildren;
+    return this;
+  }
+
   @Override
   public GetNamespacesResponse get() throws NessieReferenceNotFoundException {
     List<ContentKey> entries;
     try {
       GetEntriesBuilder getEntries = api.getEntries().refName(refName).hashOnRef(hashOnRef);
 
+      String filter = null;
       if (namespace != null && !namespace.isEmpty()) {
         String nsName = namespace.name();
-        getEntries.filter(
-            String.format(
-                "entry.encodedKey == '%s' || entry.encodedKey.startsWith('%s.')", nsName, nsName));
+        filter =
+            format(
+                "entry.encodedKey == '%s' || entry.encodedKey.startsWith('%s.')", nsName, nsName);
+        if (onlyDirectChildren) {
+          filter =
+              format(
+                  "size(entry.keyElements) == %d && (%s)",
+                  namespace.getElements().size() + 1, filter);
```

Comment:
strictly speaking the `entry.encodedKey == '%s'` filter clause is redundant in this case. Anything it matches will be excluded by the size clause.